### PR TITLE
OUT 252 Implement Delete Task in UI

### PR DESCRIPTION
**Pull request by aatbip**
### Tasks

- [Implement Delete Task in UI](https://linear.app/copilotplatforms/issue/OUT-252/implement-delete-task-in-ui)

### Changes

- [x] Adds `ListBtn` and `MoreBtn` components
- [x] Adds `taskDetailsSlice` to handle states in details page
- [x] Adds `ConfirmDeleteUI` layout that can be used for delete confirmation throughout the app
- [x] Adds delete operation in the details page and integrated delete API

### Testing Criteria

- [LOOM](https://www.loom.com/share/9e72f5780d9b4ab989b687d6b98619ff?sid=43b36f71-0e70-4b78-b6a2-fa8212983313)

### Notes

- This branch is dependent on #18. 
- Merge this branch against main after #15 is merged.

### CODE REVIEW

File: src/app/detail/ui/MenuBox.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import { ListBtn } from '@/components/buttons/ListBtn'
+import { MoreBtn } from '@/components/buttons/MoreBtn'
+import { Box, Popper } from '@mui/material'
+import { useState } from 'react'
+import { ClickAwayListener } from '@mui/base'
+import store from '@/redux/store'
+import { setShowConfirmDeleteModal } from '@/redux/features/taskDetailsSlice'
+
+export const MenuBox = () => {
```
Comment (makeitraina) [inline src/app/detail/ui/MenuBox.tsx:11]: is this supposed to be a generic component? It doesn't seem generic given that it has an explicit Delete menu item. The options should probably be passsed in as props right?

File: src/hooks/useHandleSelectorComponent.tsx [hunk 2]
```diff
@@ -19,7 +19,20 @@ export const useHandleSelectorComponent = ({ item, type }: { item: unknown; type
 
     if (type === SelectorType.ASSIGNEE_SELECTOR) {
       store.dispatch(setCreateTaskFields({ targetField: 'assigneeId', value: (item as IAssigneeCombined)?.id }))
-      store.dispatch(setCreateTaskFields({ targetField: 'assigneeType', value: (item as IAssigneeCombined)?.type }))
+      const assigneeType = (item as IAssigneeCombined).type
```
Comment (makeitraina) [inline src/hooks/useHandleSelectorComponent.tsx:22]: why do we need this casting?

**Review by makeitraina — APPROVED** [review]
generally this looks good. The loom video with testing was also helpful.

I think there is broader question about revisting the redux architecture here, but we can have that conversation separately.

File: src/app/detail/ui/MenuBox.tsx (hunk 1 above)
Comment (aatbip) [reply src/app/detail/ui/MenuBox.tsx:11]: This is not a generic component. Inside the `ui` folders of every pages, I am placing the client components with `use client` directive that are only useful for that particular page.

File: src/hooks/useHandleSelectorComponent.tsx (hunk 2 above)
Comment (aatbip) [reply src/hooks/useHandleSelectorComponent.tsx:22]: MUI Autocomplete component expects `options` with type `unknown` and so for the `useHandleSelectorComponent` hook also I have given the type for the `item` as unknown. 
This type casting are needed to cast the type for the `unknown` `item` type.

**Comment by aatbip** [comment]
Merging this branch for now. If anything is required to change, we will perform in separate PR. 
cc: @makeitraina @rrojan @sazanrjb

**Comment by rrojan** [comment]
Please fix merge conflicts @aatbip

**Pull request merged**